Remote Habits/View/RHLoginViewController.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  RHLoginViewController.swift
+//  Remote Habits Mobile App
+//
+//  Created by Amandeep Kaur on 25/11/21.
+//
+
+import UIKit
+import SkyFloatingLabelTextField
+
+
+class RHLoginViewController: RHBaseViewController {
+
+    // MARK: - --OUTLETS--
+    @IBOutlet weak var userNameInput: SkyFloatingLabelTextFieldWithIcon!
+    @IBOutlet weak var emailInput: SkyFloatingLabelTextFieldWithIcon!
+    @IBOutlet weak var guestButton: UIButton!
+    @IBOutlet weak var loginButton: UIButton!
+    @IBOutlet weak var mainView: UIView!
+
+    // MARK: - --VARIABLES--
+    var textFields: [SkyFloatingLabelTextFieldWithIcon] = []
+    
+    // MARK: - --LIFECYCLE METHODS--
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        
+        addLoginBackground()
+        textFields = [userNameInput, emailInput]
+        setUpTextFields()
+        setupButtons()
+        setUpMainView()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+
+    // MARK: - --FUNCTIONS--
+    func setupButtons() {
+        
+        // Buttons
+        loginButton.isEnabled = false
+        loginButton.titleLabel?.font = RHFont.SFProTextSemiBoldMedium
+        loginButton.setTitleColor(UIColor.gray, for: .disabled)
+        loginButton.setTitleColor(UIColor.white, for: .normal)
+    }
+    
+    func setUpTextFields() {
+        for field in textFields {
+            
+            switch(field) {
+            case userNameInput :
+                field.iconText = "\u{f007}"
+            case emailInput:
+                field.iconText = "\u{f0e0}"
+                field.errorColor = UIColor.red
+            default:
+                break
+            }
+            
+            field.iconFont = RHFont.AwesomeFontLarge
+            field.tintColor = RHColor.LabelGray
+            field.lineColor = RHColor.LineGray
+            field.selectedTitleColor = RHColor.LabelGray
+            field.selectedLineColor = RHColor.SelectedLineGray
+            field.delegate = self
+        }
+    }
+    
+    func setUpMainView() {
+        mainView.layer.cornerRadius = 13
+    }
+    
+    func validateName(_ name : String) -> Bool {
+        return name.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines) != RHConstants.kEmptyValue
+    }
+    
+    func validateEmail(_ email: String) -> Bool {
+        
+        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+
+        let emailPred = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
+        return emailPred.evaluate(with: email)
+    }
+    
+    func route(withParam isGuestLogin : Bool = true) {
```

Comment:
Not sure why param is named `withParam`? 

```suggestion
    func route(isGuestLogin : Bool = true) {
```

Reply:
This function has been revamped. Will be updating with next PR.